src/sentry/integrations/jira/integration.py

```diff
@@ -639,6 +639,21 @@ def get_create_issue_config(self, group, **kwargs):
                 field["choices"] = self.make_choices(client.get_versions(meta["key"]))
             elif field["name"] == "labels":
                 field["default"] = defaults.get("labels", "")
+            elif field["name"] == "reporter":
+                reporter_id = defaults.get("reporter", "")
+                if not reporter_id:
+                    continue
+                try:
+                    reporter_info = client.get_user(reporter_id)
+                except ApiError as exc:
+                    self.get_logger().exception(six.text_type(exc))
+                    continue
+                reporter_tuple = client.format_user(reporter_info)
```

Comment:
@saifelse I think it would make more sense to put the `format_user` functionality inside of `get_user` and call it something like `get_formatted_user` as I would usually expect a function on the client to make a network request.

Reply:
@scefali : Ah, so I had pulled out `JiraSearchEndpoint._get_formatted_user` into `JiraApiClient` so that it could be used both here and in `JiraSearchEndpoint`.

Pattern matching off other integrations, I could introduce a:

```py
# src/sentry/integrations/jira/utils.py
def build_user_choice(user_response, user_id_field):
    """
    Build an (id, label) tuple from the given the Jira REST API User resource,
    or return None if a tuple could not be built.
    """
    if user_id_field not in user_response:
        return None

    # The name field can be blank in jira-cloud, and the id_field varies by
    # jira-cloud and jira-server
    name = user_response.get("name", "")
    email = user_response.get("emailAddress")

    display = "%s %s%s" % (
        user_response.get("displayName", name),
        "- %s " % email if email else "",
        "(%s)" % name if name else "",
    )
    return user_response[user_id_field], display.strip()
```

thoughts?

Reply:
Went with this approach ^

Reply:
@saifelse looks good to me. Let me pull down and play around with the branch a bit.